rotel_python_processor_sdk/src/py/metrics.rs

```diff
@@ -0,0 +1,2170 @@
+            Ok(None)
+        }
+    }
+
+    #[setter]
+    fn set_value(&mut self, value: Option<ExemplarValue>) -> PyResult<()> {
+        let mut inner = self.inner.lock().map_err(handle_poison_error)?;
+        if value.is_none() {
+            inner.value = None;
+            return Ok(());
+        }
+        let v = value.unwrap();
+        match v {
+            ExemplarValue::AsDouble(f) => inner.value = Some(RExemplarValue::AsDouble(f)),
+            ExemplarValue::AsInt(i) => inner.value = Some(RExemplarValue::AsInt(i)),
+        }
+        Ok(())
+    }
+}
+
+// --- PyO3 Bindings for RExemplarValue (Enum) ---
+#[pyclass]
+#[derive(Clone)]
+pub enum ExemplarValue {
+    AsDouble(f64),
+    AsInt(i64),
+}
+
+// --- PyO3 Bindings for RAggregationTemporality (Enum) ---
+#[pyclass]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub enum AggregationTemporality {
+    Unspecified = 0,
+    Delta = 1,
+    Cumulative = 2,
+}
+
+#[pymethods]
+impl AggregationTemporality {
+    #[new]
+    fn new() -> PyResult<Self> {
+        Ok(AggregationTemporality::Unspecified)
+    }
+
+    pub fn as_str_name(&self) -> &'static str {
+        match self {
+            Self::Unspecified => "AGGREGATION_TEMPORALITY_UNSPECIFIED",
+            Self::Delta => "AGGREGATION_TEMPORALITY_DELTA",
+            Self::Cumulative => "AGGREGATION_TEMPORALITY_CUMULATIVE",
+        }
+    }
+}
+
+impl From<i32> for AggregationTemporality {
+    fn from(value: i32) -> Self {
+        match value {
+            1 => AggregationTemporality::Delta,
+            2 => AggregationTemporality::Cumulative,
+            _ => AggregationTemporality::Unspecified,
+        }
+    }
+}
+impl From<AggregationTemporality> for i32 {
+    fn from(value: AggregationTemporality) -> i32 {
+        match value {
+            AggregationTemporality::Unspecified => 3,
+            AggregationTemporality::Delta => 1,
+            AggregationTemporality::Cumulative => 2,
+        }
+    }
+}
+
+// --- PyO3 Bindings for RDataPointFlags (Enum) ---
+#[pyclass]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub enum DataPointFlags {
+    DoNotUse = 0,
+    NoRecordedValueMask = 1,
+}
+
+#[pymethods]
+impl DataPointFlags {
+    #[new]
+    fn new() -> PyResult<Self> {
+        Ok(DataPointFlags::DoNotUse)
+    }
+
+    pub fn as_str_name(&self) -> &'static str {
+        match self {
+            Self::DoNotUse => "DATA_POINT_FLAGS_DO_NOT_USE",
+            Self::NoRecordedValueMask => "DATA_POINT_FLAGS_NO_RECORDED_VALUE_MASK",
+        }
+    }
+
+    #[getter]
+    fn value(&self) -> PyResult<u32> {
+        match self {
+            DataPointFlags::DoNotUse => Ok(0),
+            DataPointFlags::NoRecordedValueMask => Ok(1),
+        }
+    }
+}
+
+impl From<u32> for DataPointFlags {
+    fn from(value: u32) -> Self {
+        match value {
+            1 => DataPointFlags::NoRecordedValueMask,
+            _ => DataPointFlags::DoNotUse,
```

Comment:
If you want these to be explicit, you could also use the TryFrom trait: https://doc.rust-lang.org/std/convert/trait.TryFrom.html